test/libsolidity/SemanticTest.cpp

```diff
@@ -420,15 +449,23 @@ void SemanticTest::printUpdatedExpectations(ostream& _stream, string const&) con
 void SemanticTest::printUpdatedSettings(ostream& _stream, string const& _linePrefix)
 {
 	auto& settings = m_reader.settings();
-	if (settings.empty() && !m_compileViaYulCanBeSet)
+	if (settings.empty() && !m_canEnableYulRun)
 		return;
 
 	_stream << _linePrefix << "// ====" << endl;
-	if (m_compileViaYulCanBeSet)
+	if (m_canEnableEwasmRun)
+	{
+		string compileViaYul = m_reader.stringSetting("compileViaYul", "");
+		if (!compileViaYul.empty())
+			_stream << _linePrefix << "// compileViaYul: " << compileViaYul << "\n";
+		_stream << _linePrefix << "// compileToEwasm: also\n";
+	}
+	else if (m_canEnableYulRun)
 		_stream << _linePrefix << "// compileViaYul: also\n";
+
 	for (auto const& setting: settings)
-		if (!m_compileViaYulCanBeSet || setting.first != "compileViaYul")
-		_stream << _linePrefix << "// " << setting.first << ": " << setting.second << endl;
+		if ((!m_canEnableYulRun && !m_canEnableEwasmRun) || setting.first != "compileViaYul")
+			_stream << _linePrefix << "// " << setting.first << ": " << setting.second << endl;
```

Comment:
This was changed but not exactly the way I suggested in https://github.com/ethereum/solidity/pull/10356#discussion_r565529135.

The loop should skip `compileToEwasm` when `m_canEnableEwasmRun` is true because the `if` above handles it in that case. The way it is now, it will print it twice in that case.